handlers/controller.js

```diff
@@ -23,6 +23,41 @@ let welcome = (req, res) => {
     res.status(StatusCode.EASTER_EGG).send('Welcome to the Greenmap-API!');
 };
 
+
+let getTopManufacturers = async(req, res) => {
+    connectDb().then(async () => {
+        if(!req.params.id) {
+            return res.status(StatusCode.PRECONDITION_FAILED).send(null);
+        }      
+        let barcode = req.params.id;
+        try {
+            let doc = await models.Product.findOne({ barcode });
+            if(doc) {
+                console.log(`found ${doc.barcode} in mongodb`);
+                if(!doc.category || doc.category == null) {
+                    console.log(`category not found for ${doc.name}`)
+                    return res.status(StatusCode.PRECONDITION_FAILED).send();
+                }
+
+                let category = doc.category[doc.category.length - 1];
+                getSimilarCategory(category, res);
+            }
+
+            if(!doc) {
```

Comment:
This can be simplified to:

```
console.error(`barcode ${barcode} not found in the database`);
return res.status(StatusCode.BAD_REQUEST).send(); 
```

You don't need to check if the document is present again here.

Reply:
The `doc` here represents the object/document that corresponding to the barcode entered, which may not be available.

However, the one inside of `getSimilarCategory`, i.e.
`if(!docs) {
        return res.status(StatusCode.BAD_REQUEST).send();
  }`
  
pertains to a different query. Here, `docs ` represents the documents that have the same category as the product represented by the barcode id.

Reply:
I'm not totally following how that relates to this. On line `35` and `36` you are basically preforming an `if() ... else if()` operation. This can be simplified to just a single `if`. If the `if` on line 35 is successful then this handler will return on line 39, otherwise it will return on line 52.

Reply:
```suggestion
let getTopManufacturers = async(req, res) => {
    connectDb().then(async () => {
        if(!req.params.id) {
            return res.status(StatusCode.PRECONDITION_FAILED).send(null);
        }      
        let barcode = req.params.id;
        try {
            let doc = await models.Product.findOne({ barcode });
            if(doc) {
                console.log(`found ${doc.barcode} in mongodb`);
                if(!doc.category || doc.category == null) {
                    console.log(`category not found for ${doc.name}`)
                    return res.status(StatusCode.PRECONDITION_FAILED).send();
                }

                let category = doc.category[doc.category.length - 1];
                getSimilarCategory(category, res);
            }
            // else
            console.log(`barcode ${barcode} not found in the database`);
            return res.status(StatusCode.BAD_REQUEST).send();
        } catch(err) {
            console.error(err);
            return res.status(StatusCode.BAD_REQUEST).send(err);
        }
    }).catch((err) => {
        console.error(err);
        return res.status(StatusCode.INTERNAL_SERVER_ERROR).send(err);
    });
};
```

Reply:
```
let getTopManufacturers = async(req, res) => {
    connectDb().then(async () => {
        if(!req.params.id) {
            return res.status(StatusCode.PRECONDITION_FAILED).send(null);
        }      
        let barcode = req.params.id;
        try {
            let doc = await models.Product.findOne({ barcode });
            if(doc) {
                console.log(`found ${doc.barcode} in mongodb`);
                if(!doc.category || doc.category == null) {
                    console.log(`category not found for ${doc.name}`)
                    return res.status(StatusCode.PRECONDITION_FAILED).send();
                }

                let category = doc.category[doc.category.length - 1];
                getSimilarCategory(category, res);
            } ```
```suggestion
            // else
            console.log(`barcode ${barcode} not found in the database`);
            return res.status(StatusCode.BAD_REQUEST).send();
            ```
```
        } catch(err) {
            console.error(err);
            return res.status(StatusCode.BAD_REQUEST).send(err);
        }
    }).catch((err) => {
        console.error(err);
        return res.status(StatusCode.INTERNAL_SERVER_ERROR).send(err);
    });
};
```

Reply:
I thought you were referring to erasing the else part of the code, because `getTopManufacturer` calls `getSimilarCategory` where they both check if the returned document(s) are found or not. Apologies for the the misunderstanding.